subcommands/keys/tuf_updates_rotate_offline_key.go

```diff
@@ -226,36 +224,21 @@ func replaceOfflineTargetsKey(
 	return &kp.signer, creds
 }
 
-func resignProdTargets(
-	factory string, root *client.AtsTufRoot, onlineTargetsId string, creds OfflineCreds,
-) (map[string][]tuf.Signature, error) {
+func resignProdTargets(factory string, signer *TufSigner) (map[string][]tuf.Signature, error) {
 	targetsMap, err := api.ProdTargetsList(factory, false)
 	if err != nil {
 		return nil, fmt.Errorf("Failed to fetch production targets: %w", err)
 	} else if targetsMap == nil {
 		return nil, nil
 	}
 
-	var signers []TufSigner
-	for _, kid := range root.Signed.Roles["targets"].KeyIDs {
-		if kid == onlineTargetsId {
-			continue
-		}
-		pub := root.Signed.Keys[kid].KeyValue.Public
-		signer, err := FindTufSigner(kid, pub, creds)
-		if err != nil {
-			return nil, fmt.Errorf("Failed to find private key for %s: %w", kid, err)
-		}
-		signers = append(signers, *signer)
-	}
-
 	signatureMap := make(map[string][]tuf.Signature)
 	for tag, targets := range targetsMap {
 		bytes, err := canonical.MarshalCanonical(targets.Signed)
 		if err != nil {
 			return nil, fmt.Errorf("Failed to marshal targets for tag %s: %w", tag, err)
 		}
-		signatures, err := SignTufMeta(bytes, signers...)
+		signatures, err := SignTufMeta(bytes, *signer)
```

Comment:
Are we sure that `signer` is not nil at this point?

Reply:
Thanks for the comment.

I need one more commit which makes `replaceOfflineRootKey` and `replaceOfflineTargetsKey` to return value types, so that I can make `resignProdTargets` accept a value type too.

This value can never be `nil`, so there is no reason to make it a reference type.

TL;DR
IIRC this is a historic baggage from the times some of us believed that having a reference makes the program faster because of not having to do the copy-on-passing-argument for the value types. It turns out the effect is quite opposite, as Golang leverages CPU registers a lot to pass small value types, making that faster. So, we unintentionally made some places of our code nil-poisoned while striving to achieve a nonexisting benefit.

Reply:
https://github.com/foundriesio/fioctl/pull/238/commits/6c1e20c1712a2ce8d4b640cdd48c07387378531a

Reply:
https://github.com/foundriesio/fioctl/pull/238/commits/6c1e20c1712a2ce8d4b640cdd48c07387378531a